include/odp/api/spec/packet_io.h

```diff
@@ -1394,6 +1397,18 @@ uint64_t odp_pktin_ts_res(odp_pktio_t pktio);
  */
 odp_time_t odp_pktin_ts_from_ns(odp_pktio_t pktio, uint64_t ns);
 
+/**
+ * Get Tx timestamp inserted for a packet if available and clear it for
+ * next timestamp.
+ *
+ * @param pktio handle identifying pktio
+ * @param ts    Pointer to odp_time_t where timestamp will be written
+ *
+ * @retval zero     Successfully returns last pkt timestamp.
+ * @retval negative error code on failure.
+ */
+int odp_pktout_ts(odp_pktio_t pktio, odp_time_t *ts);
```

Comment:
I was thinking if this is a good API instead of defining an event and/or making it possible to have multiple timestamped packets in flight in the same pktio (as supported by some HW). But maybe this is better as it is simple.

Reply:
I also think, since PTP packets are not that frequent and also we need them to work even when application cannot support schedule mode, this API is simple and sufficient for the usecase.
We can also extended this in future to generate events if any vendor hardware supports it.

Reply:
How application is able to be sure which (outgoing) packet the timestamp refers to. I think application would need to mark the packet that requests a time stamp during output. E.g.

/**
 *  Request packet output to record a time stamp
 *
 * Time stamp recorded when this packet is output on the interface. Time stamp 
 * can be retrieved with odp_pktout_ts().
*/
int odp_packet_ts_request(odp_packet_t pkt);

Reply:
odp_pktout_ts() is a bit short / not very descriptive name. Would e.g. odp_pktout_ts_poll() or odp_pktout_ts_read() tell that this function is used asynchronously to poll the interface after packet output. It's part of normal operation that application needs to poll for an implementation specific time before the time stamp is available.

Reply:
Ok. Will change to odp_pktout_ts_read() then.